### PR TITLE
Expire tokens when created_at is not available

### PR DESCRIPTION
**Pull request by juanibiapina**
Solve #47

### CODE REVIEW

**Comment by juanibiapina** [comment]
Closing because project seems unmaintained.

File: lib/devise/token_authenticatable/strategy.rb [hunk 1]
```diff
@@ -37,6 +37,8 @@ def authenticate!
         return fail(:invalid_token) unless resource
 
         unless token_expires_in.blank?
+          return fail(:expired_token) if resource.authentication_token_created_at.nil?
```
Comment (baschtl) [inline lib/devise/token_authenticatable/strategy.rb]: Could you extract the conditions into a separate method like so:

```ruby
def authenticate!
  # ...
  unless token_expires_in.blank?
    return fail(:expired_token) if authentication_token_expired?(resource.authentication_token_created_at)
  end
  # ...
end

private

def authentication_token_expired?(token_created_at)
  token_created_at.nil? || Time.now > (token_created_at + token_expires_in.to_i)
end
```

?

File: spec/factories/user.rb [hunk 2]
```diff
@@ -28,5 +28,10 @@
       authentication_token { SecureRandom.hex }
       authentication_token_created_at { Time.now - 1.day }
     end
+
+    trait :without_created_at do
```
Comment (baschtl) [inline spec/factories/user.rb]: Please disambiguate here and call the trait `without_authentication_token_created_at`.

File: spec/requests/devise/token_authenticatable/strategy_spec.rb [hunk 3]
```diff
@@ -341,6 +341,48 @@
...
+        end
+      end
+
+      it "should not authenticate user" do
+        swap Devise::TokenAuthenticatable, token_expires_in: 1.hour do
+          sign_in_as_new_user_with_token(use: :without_created_at)
+
+          expect(warden).to_not be_authenticated(:user)
+        end
+      end
```
Comment (baschtl) [inline spec/requests/devise/token_authenticatable/strategy_spec.rb]: The context `"through params"` should end here, no?

**Pull request closed without merging**